src/test/java/org/threeten/extra/chrono/TestInternationalFixedChronology.java

```diff
@@ -0,0 +1,794 @@
+            final int expectedMonth,
+            final int expectedDom) {
+        assertEquals (InternationalFixedDate.of (year, month, dom).with (field, value), InternationalFixedDate.of (expectedYear, expectedMonth, expectedDom));
+    }
+
+    @Test (expectedExceptions = UnsupportedTemporalTypeException.class)
+    public void test_with_TemporalField_unsupported () {
+        InternationalFixedDate.of (2012, 6, 28).with (ChronoField.MINUTE_OF_DAY, 0);
+    }
+
+    //-----------------------------------------------------------------------
+    // InternationalFixedDate.with(TemporalAdjuster)
+    //-----------------------------------------------------------------------
+    @Test
+    public void test_adjust_toLastDayOfMonth () {
+        InternationalFixedDate base = InternationalFixedDate.of (2012, 6, 23);
+        InternationalFixedDate test = base.with (TemporalAdjusters.lastDayOfMonth ());
+        assertEquals (test, InternationalFixedDate.of (2012, 6, 29));
+
+        base = InternationalFixedDate.of (2012, 13, 2);
+        test = base.with (TemporalAdjusters.lastDayOfMonth ());
+        assertEquals (test, InternationalFixedDate.of (2012, 13, 29));
+    }
+
+    //-----------------------------------------------------------------------
+    // InternationalFixedDate.with(Local*)
+    //-----------------------------------------------------------------------
+    @Test
+    public void test_adjust_toLocalDate () {
+        InternationalFixedDate date = InternationalFixedDate.of (2000, 1, 4);
+        InternationalFixedDate test = date.with (LocalDate.of (2012, 7, 14));
+        assertEquals (test, InternationalFixedDate.of (2012, 7, 27));
+    }
+
+    @Test (expectedExceptions = DateTimeException.class)
+    public void test_adjust_toMonth () {
+        InternationalFixedDate date = InternationalFixedDate.of (2000, 1, 4);
+        date.with (Month.APRIL);
+    }
+
+    //-----------------------------------------------------------------------
+    // LocalDate.with(InternationalFixedDate)
+    //-----------------------------------------------------------------------
+    @Test
+    public void test_LocalDate_adjustToInternationalFixedDate () {
+        InternationalFixedDate date = InternationalFixedDate.of (2012, 6, 15);
+        LocalDate test = LocalDate.MIN.with (date);
+        assertEquals (test, LocalDate.of (2012, 6, 4));
+    }
+
+    @Test
+    public void test_LocalDateTime_adjustToInternationalFixedDate () {
+        InternationalFixedDate date = InternationalFixedDate.of (2012, 6, 15);
+        LocalDateTime test = LocalDateTime.MIN.with (date);
+        assertEquals (test, LocalDateTime.of (2012, 6, 4, 0, 0));
+    }
+
+    //-----------------------------------------------------------------------
+    // InternationalFixedDate.plus
+    //-----------------------------------------------------------------------
+    @DataProvider (name = "plus")
+    Object[][] data_plus () {
+        return new Object[][] {
```

Comment:
What happens when you add a week to the leap-day/year-day?  Remember, they're not part of the normal week (or month, for that matter) - Discordian goes into this some, too.